api/v1alpha4/cluster_webhook.go

```diff
@@ -86,6 +103,111 @@ func (c *Cluster) validate() error {
 		)
 	}
 
+	// If the Cluster uses a managed topology.
+	if c.Spec.Topology != nil {
```

Comment:
Right after we get here, we should check that the feature gate is enabled or return an error right away.

Reply:
done